assets/js/modules/thank-with-google/components/setup/SetupMain.js

```diff
@@ -70,6 +73,27 @@ export default function SetupMain( { finishSetup } ) {
 	// Reset all fetched data when user re-focuses window.
 	useRefocus( reset, 15000 );
 
+	const viewContext = useViewContext();
+
+	useEffect( () => {
+		if ( currentPublication === undefined ) {
+			return;
+		}
+		if ( currentPublication === null ) {
+			trackEvent(
+				`${ viewContext }_thank-with-google`,
+				'receive_publication_state',
+				ONBOARDING_STATE_NO_ACCOUNT
+			);
+		} else {
+			trackEvent(
+				`${ viewContext }_thank-with-google`,
+				'receive_publication_state',
+				currentPublication.onboardingState
+			);
+		}
```

Comment:
This might actually be a rare case where I'm pro-ternary expression, because a lot of this is duplicated and only one variable changes based on the `if` condition.

I think:

```js
		// Don't track an event if the current publication is still loading.
		if ( currentPublication === undefined ) {
			return;
		}

		trackEvent(
			`${ viewContext }_thank-with-google`,
			'receive_publication_state',
			currentPublication === null
				? ONBOARDING_STATE_NO_ACCOUNT
				: currentPublication.onboardingState
		);
```

might be easier to read; a rarity for ternary expressions!